pkg/cgroups/cgroupblkio.go

```diff
@@ -0,0 +1,168 @@
+	return OciBlockIOParameters{
+		Weight: -1,
+	}
+}
+
+// NewOciWeightDeviceParametes creates new OciWeightDeviceParameters instance.
+func NewOciWeightDeviceParametes() OciWeightDeviceParameters {
+	return OciWeightDeviceParameters{
+		Major:  -1,
+		Minor:  -1,
+		Weight: -1,
+	}
+}
+
+// NewOciRateDeviceParameters creates new OciRateDeviceParameters instance.
+func NewOciRateDeviceParameters() OciRateDeviceParameters {
+	return OciRateDeviceParameters{
+		Major: -1,
+		Minor: -1,
+		Rate:  -1,
+	}
+}
+
+// GetBlkioDir returns the cgroups blkio controller directory.
+func GetBlkioDir() string {
+	return blkioCgroupDir
+}
+
+// SetBlkioParameters writes OCI BlockIO parameters to files in cgroups blkio contoller directory.
+func SetBlkioParameters(cgroupsDir string, blockIO OciBlockIOParameters) error {
+	log.Debug("configuring cgroups blkio controller in directory %#v with parameters %+v", cgroupsDir, blockIO)
+	var errors *multierror.Error
+	if blockIO.Weight >= 0 {
+		errors = multierror.Append(errors, writeToFileInDir(cgroupsDir, blkioWeightFiles, strconv.FormatInt(blockIO.Weight, 10)))
+	}
+	for _, weightDevice := range blockIO.WeightDevice {
+		errors = multierror.Append(errors, writeDevValueToFileInDir(cgroupsDir, blkioWeightDeviceFiles, weightDevice.Major, weightDevice.Minor, weightDevice.Weight))
+	}
+	for _, rateDevice := range blockIO.ThrottleReadBpsDevice {
+		errors = multierror.Append(errors, writeDevValueToFileInDir(cgroupsDir, blkioThrottleReadBpsFiles, rateDevice.Major, rateDevice.Minor, rateDevice.Rate))
+	}
+	for _, rateDevice := range blockIO.ThrottleWriteBpsDevice {
+		errors = multierror.Append(errors, writeDevValueToFileInDir(cgroupsDir, blkioThrottleWriteBpsFiles, rateDevice.Major, rateDevice.Minor, rateDevice.Rate))
+	}
+	for _, rateDevice := range blockIO.ThrottleReadIOPSDevice {
+		errors = multierror.Append(errors, writeDevValueToFileInDir(cgroupsDir, blkioThrottleReadIOPSFiles, rateDevice.Major, rateDevice.Minor, rateDevice.Rate))
+	}
+	for _, rateDevice := range blockIO.ThrottleWriteIOPSDevice {
+		errors = multierror.Append(errors, writeDevValueToFileInDir(cgroupsDir, blkioThrottleWriteIOPSFiles, rateDevice.Major, rateDevice.Minor, rateDevice.Rate))
+	}
+	return errors.ErrorOrNil()
+}
+
+// writeDevValueToFileInDir writes MAJOR:MINOR VALUE to the first existing file under baseDir
+func writeDevValueToFileInDir(baseDir string, filenames []string, major, minor, value int64) error {
+	content := fmt.Sprintf("%d:%d %d", major, minor, value)
+	return writeToFileInDir(baseDir, filenames, content)
+}
+
+// writeToFileInDir writes content to the first existing file in the list under baseDir.
+func writeToFileInDir(baseDir string, filenames []string, content string) error {
+	var errors *multierror.Error
+	// Returns list of errors from writes, list of single error due to all filenames missing or nil on success.
+	for _, filename := range filenames {
+		filepath := baseDir + "/" + filename
```

Comment:
https://golang.org/pkg/path/filepath/#Join

Reply:
Thanks, done in next push.